applications/serial_lte_modem/src/slm_at_socket.c

```diff
@@ -238,12 +240,6 @@ static int do_secure_socket_open(int peer_verify)
 	return 0;
 
 error_exit:
-#if defined(CONFIG_SLM_NATIVE_TLS)
-	if (sock.type == SOCK_STREAM && sock.sec_tag != INVALID_SEC_TAG) {
-		(void)slm_tls_unloadcrdl(sock.sec_tag);
-		sock.sec_tag = INVALID_SEC_TAG;
-	}
-#endif
```

Comment:
In case the credentials loading succeeded but the socket creation failed later, the just-loaded credentials will be kept and in the case of there being 2, the next cred to be used will be the other one, that may already be in use, while this one will remain unused.
I imagine this could cause some issues, and that it may be better to provide an unload function to be called as before.
What do you think?

Reply:
I would like to avoid providing an unload function for this. The scenario you are describing would require 3 different sec tags in use with multiple servers / clients. Generally I believe that 1 buffer would be enough for most of the cases.